api/dataset.go

```diff
@@ -680,6 +680,75 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 	log.Debug("get dimension options", log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
 }
 
+func (api *DatasetAPI) getMetaData(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	datasetID := vars["id"]
+	edition := vars["edition"]
+	version := vars["version"]
+
+	// get dataset document
+	datasetDoc, err := api.dataStore.Backend.GetDataset(datasetID)
+	if err != nil {
+		log.Error(err, log.Data{"dataset_id": datasetID, "edition": edition, "version": version})
+		handleErrorType(versionDocType, err, w)
+		return
+	}
+
+	// Default state to published
+	state := models.PublishedState
+
+	// if request is authenticated then access resources of state other than published
+	if r.Header.Get(internalToken) == api.internalToken {
+		state = ""
+	}
+
+	if state == models.PublishedState {
```

Comment:
could remove the 7 lines above and change this if to `r.Header != api.internalToken` i think? just `var state string` before and on line 711 before leaving the if set state to `datasetDoc.Current.State` (or models.PublishedState)

Reply:
It would have to be `models.PublishedState` as there may not be a current sub document. I will change it :)